warehouse/query-core/src/main/java/datawave/query/jexl/visitors/ExecutableDeterminationVisitor.java

```diff
@@ -594,390 +405,1040 @@ public Object visit(ASTLTNode node, Object data) {
-        }
+        writeNodeToOutput(node, data, state);
         return state;
     }
     
+    /**
+     * Returns {@link STATE#IGNORABLE}
+     *
+     * @param node
+     *            the node
+     * @param data
+     *            the data
+     * @return the state
+     */
     @Override
     public Object visit(ASTVar node, Object data) {
         STATE state = STATE.IGNORABLE;
-        if (output != null) {
-            output.writeLine(data + node.toString() + " -> " + state);
-        }
+        writeNodeToOutput(node, data, state);
         return state;
     }
     
+    /**
+     * Returns {@link STATE#IGNORABLE}
+     *
+     * @param node
+     *            the node
+     * @param data
+     *            the data
+     * @return the state
+     */
     @Override
     public Object visit(ASTNumberLiteral node, Object data) {
         STATE state = STATE.IGNORABLE;
-        if (output != null) {
-            output.writeLine(data + node.toString() + " -> " + state);
+        writeNodeToOutput(node, data, state);
+        return state;
+    }
+    
+    /**
+     * All children must be must be executable for the specified node to be executable. Used for expressions, scripts, and or nodes. Returns a state based on
+     * the states retrieved for each child of the specified node based on the following cases, in priority order:
+     * <ol>
+     * <li>
+     * If no states are found, {@link STATE#PARTIAL} is returned.</li>
+     * <li>
+     * If one unique state is found, that state is returned.</li>
+     * <li>
+     * If two unique states are found, and one is {@link STATE#IGNORABLE}, the other state is returned.</li>
+     * <li>
+     * If three or more unique states are found and one is {@link STATE#ERROR}, then {@link STATE#ERROR} is returned.</li>
+     * <li>
+     * If three or more unique states are found and one is {@link STATE#PARTIAL}, then {@link STATE#PARTIAL} is returned.</li>
+     * <li>
+     * If only {@link STATE#EXECUTABLE}, {@link STATE#NON_EXECUTABLE}, and {@link STATE#IGNORABLE} are found, {@link STATE#PARTIAL} is returned.</li>
+     * </ol>
+     *
+     * @param node
+     *            the node
+     * @param data
+     *            the data
+     * @return the state
+     */
+    protected STATE allOrNone(JexlNode node, Object data) {
+        // Find the states of each child, sorting them in all-or-none priority.
+        final SortedSet<STATE> states = new TreeSet<>(allOrNoneComparator);
+        readStatesFromChildren(node, data + PREFIX, states);
+        
+        if (log.isTraceEnabled()) {
+            log.trace("node:" + PrintingVisitor.formattedQueryString(node));
+            log.trace("states are:" + states);
+        }
+        
+        // If no states were found, or both EXECUTABLE and NON_EXECUTABLE were found, return PARTIAL.
+        final STATE state;
+        if (states.size() == 0 || (states.size() > 2 && states.first() == STATE.EXECUTABLE)) {
+            state = STATE.PARTIAL;
```

Comment:
This behavior isn't different from before, but a STATE.PARTIAL on no states seems wrong

Reply:
Perhaps NON_EXECUTABLE would make more sense?